### ISSUE TITLE
SmartNulls should identify the arguments that the stubbed method was invoked with

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#225]
```
I've started mocking with RETURNS_SMART_NULLS. It is very useful and I like the 
fact that it can be configured with the @Mock annotation. I agree with the 
comment that it should be the default in Mockito 2.0. Returning null values 
from mocked methods encourages writing code that is designed to handle nulls, 
which IMO is a code smell.

One thing that would be really nice is if the arguments of the stubbed method 
were retained and then printed when the smart null is used. The reported 
message is typically correct: "Because this method was *not* stubbed 
correctly:", but this is generally because I've made some mistake in matching 
the parameters of the stubbed method. If I could see the arguments that were 
used in the invocation it would mean that I wouldn't need to debug into the 
method to check them.
```

Original issue reported on code.google.com by `exort...@gmail.com` on 15 Oct 2010 at 2:18

**Comment by GoogleCodeExporter** [comment]
```
I agree
```

Original comment by `szcze...@gmail.com` on 16 Oct 2010 at 10:51
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Hi, I've just implemented this issue on my clone (lucasmrtuner-mockito).

I don't know how to send an official pull request here, so I'm asking for it 
here :)

By the way, how is the right way to send a pull request?

[]'s
```

Original comment by `lucasmrt...@gmail.com` on 5 Nov 2010 at 9:06

**Comment by GoogleCodeExporter** [comment]
```
cool!

Thanks, I'll to pull it tomorrow. This is fine for pull request.

Thanks :)
```

Original comment by `szcze...@gmail.com` on 6 Nov 2010 at 11:25

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 98bf479afd.
```

Original comment by `szcze...@gmail.com` on 7 Nov 2010 at 12:07
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 3 Jul 2011 at 12:43
- Added labels: **Milestone-Release1.9-rc1**
